packages/platform-browser/src/IndexedDbDatabase.ts

```diff
@@ -0,0 +1,263 @@
+              )
+            )
+          }
+
+          request.onerror = (event) => {
+            const idbRequest = event.target as IDBRequest<IDBDatabase>
+
+            resume(
+              Effect.fail(
+                new IndexedDbDatabaseError({
+                  reason: "OpenError",
+                  cause: idbRequest.error
+                })
+              )
+            )
+          }
+
+          request.onupgradeneeded = (event) => {
+            const idbRequest = event.target as IDBRequest<IDBDatabase>
+            const database = idbRequest.result
+            const transaction = idbRequest.transaction
+            oldVersion = event.oldVersion
+
+            if (transaction === null) {
+              return resume(
+                Effect.fail(
+                  new IndexedDbDatabaseError({
+                    reason: "TransactionError",
+                    cause: null
+                  })
+                )
+              )
+            }
+
+            transaction.onabort = (event) => {
+              resume(
+                Effect.fail(
+                  new IndexedDbDatabaseError({
+                    reason: "Aborted",
+                    cause: event
+                  })
+                )
+              )
+            }
+
+            transaction.onerror = (event) => {
+              resume(
+                Effect.fail(
+                  new IndexedDbDatabaseError({
+                    reason: "TransactionError",
+                    cause: event
+                  })
+                )
+              )
+            }
+
+            const effect = Effect.forEach(migrations.slice(oldVersion), (untypedMigration) => {
+              if (untypedMigration._tag === "Migration") {
+                const migration = untypedMigration as IndexedDbMigration.IndexedDbMigration.AnyMigration
+                const fromApi = internal.makeTransactionProto({
+                  database,
+                  IDBKeyRange,
+                  tables: migration.fromVersion.tables,
+                  transaction
+                })
+                const toApi = internal.makeTransactionProto({
+                  database,
+                  IDBKeyRange,
+                  tables: migration.toVersion.tables,
+                  transaction
+                })
+                return migration.execute(fromApi, toApi)
+              } else if (untypedMigration._tag === "Initial") {
+                const migration = untypedMigration as IndexedDbMigration.IndexedDbMigration.AnyInitial
+                const api = internal.makeTransactionProto({
+                  database,
+                  IDBKeyRange,
+                  tables: migration.version.tables,
+                  transaction
+                })
+                return migration.execute(api)
+              }
+              return Effect.dieMessage("Invalid migration")
+            }, { discard: true })
+
+            Runtime.runPromise(runtime)(effect.pipe(
```

Comment:
Do you need to use runSync here?

Reply:
`runSync` seems to block `onupgradeneeded` from exiting (the tests timeout), while `runPromise` doesn’t